build_sys_common.rs

```diff
@@ -0,0 +1,31 @@
+use std::env::var;
+
+// Attempts to discover which library file to use and where it is located.
+//
+// If the `cfg(target_env)` is set to "MSVC" then it will use the MSVC
+// libraries, otherwise it will assume "GNU" and use the GCC libraries.
+//
+// If the environment variable `CSFML_HOME` is set, then it will look there
+// for the libraries, otherwise it defaults to `.`.
+//
+// It assumes that CSFML is simply unzipped - as is - into a folder, and the
+// libraries exist in: `$CSFML_HOME/lib/[msvc|gcc]/*.lib`.
+//
+pub fn lib_path(file: &str) -> String {
+    let lib = if cfg!(target_env = "msvc") {
+        format!("csfml-{}", file)
+    } else {
+        format!("libcsfml-{}", file)
```

Comment:
Oh, yeah, I realize what's up now.

Under the GNU toolchain, library files are always prefixed with `lib`, but the `lib` prefix is not included when supplying the library name to the linker. It's just how the GNU linker works I guess.

So for example, the library is named `libfoo.a`, but the library name you pass to the linker is just `foo`.

This is how it works on Unix, but I believe this is how it works on Windows under the GNU toolchain as well. Did you test this on Windows with the GNU toolchain? It should fail to link with `libcsfml-*`, but should succeed with `csfml-*`.